### PR TITLE
[docs] Update to scripts building index.md

### PR DESCRIPTION
**Pull request by MKhalusova**
The `docs/index.md` file currently contains two auto-generated parts: the list of models (same as in README), and a table of models with supported frameworks. Due to the number of models available in transformers (200+), the list and the table have become quite large, and there have been internal discussions about removing the list of models from the `index.md`. 

This PR adds the following changes: 
- removes the autogenerated model list from the `index.md` and updates the script so it's no longer added
- modifies the script that generates the table to make model names links to corresponding model_doc. 

The model lists in the main README and localized READMEs remain as is.

### CODE REVIEW

**Review by stevhliu — COMMENTED** [review]
Super nice, I think it's a lot cleaner like this! ✨

It looks like some models (like Llama 2, Flan T5, DiT) are missing though.

File: utils/check_table.py
Comment (ArthurZucker) [inline utils/check_table.py:1]: I think it is a good time to check model on the table has a corresponding `model.md` and vice-versa!

File: utils/check_copies.py
Comment (ArthurZucker) [inline utils/check_copies.py:1]: Nice!

File: docs/source/en/index.md
Comment (ArthurZucker) [inline docs/source/en/index.md:1]: A loooooot better indeed 😉

**Review by ArthurZucker — APPROVED** [review]
Very clean thanks a lot 🤗

**Comment by MKhalusova** [comment]
> Super nice, I think it's a lot cleaner like this! ✨
> 
> It looks like some models (like Llama 2, Flan T5, DiT) are missing though.

The reason they are missing is that the table is based on the model classes. And in case of the models you mentioned, Llama 2 uses the same implementation as Llama, Flan T5 uses the same implementation as T5, and DiT's architecture is equivalent to that of BEiT. 
I can check if I can automatically find model like this, if not, I'll add them as "special case" constants.

**Comment by ArthurZucker** [comment]
Yep, I think these models should be part of the table and we should make sure we can easily add more models there that share the same modeling file!

**Comment by MKhalusova** [comment]
I've added a dict of models that use the same config as some "base" model. This way the table is complete. 
I couldn't find this mapping anywhere else, so it's hardcoded here. 
If these models were included in `transformers_module.models.auto.configuration_auto.CONFIG_MAPPING_NAMES`, it would not need to be hardcoded here, however, that dictionary is used in so many places, I'm hesitant to modify it.

File: utils/check_table.py
Comment (MKhalusova) [reply utils/check_table.py:1]: I believe `check_copies.py` checks consistency between `model.md` and the list of models in `MODEL_NAMES_MAPPING`. Here, I use the same `MODEL_NAMES_MAPPING`, so I'm not sure one more check here is necessary.

**Review by stevhliu — APPROVED** [review]
Great job, and thanks for completing the table!

File: utils/check_table.py
Comment (MKhalusova) [reply utils/check_table.py:1]: Let me know if you think it's good to merge as is. Sorry about the ping @ArthurZucker :)

**Pull request merged**